### PR TITLE
Feature/local server

### PR DESCRIPTION
**Pull request by mooyoul**
# What's Changes

### [Screencast (Asciinema)](https://asciinema.org/a/f3tCW8Se4HRpHFyJBiNcOkPnJ)

Now corgi shipped with its local server to support more easier development.
basically, local server is separated to two parts: `corgi-local.ts` and `corgi-local-cli.ts`

`corgi-local.ts` exports `createServer` function. this function takes corgi handler, and returns instance of [`http.Server` class](https://nodejs.org/docs/latest-v6.x/api/http.html#http_class_http_server) which is wrapped by [zeit/micro](https://github.com/zeit/micro).
so corgi developers can extend local server if needed.

`corgi-local-cli.ts` is simple cli program.
it just takes some configurations from its command arguments, creates new server using `createServer` function from `corgi-local`, and listen on specified port. that's all.

`corgi-local-cli.ts` is exposed to `corgi-local` executable binary.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,26 +27,33 @@
...
-    "@types/underscore.string": "0.0.30",
-    "joi-to-json-schema": "^3.2.0",
     "aws-xray-sdk-core": "^1.1.2",
+    "bluebird": "^3.5.1",
+    "commander": "^2.11.0",
+    "debug": "^3.1.0",
     "joi": "^10.5.2",
+    "joi-to-json-schema": "^3.2.0",
     "lodash": "^4.17.4",
+    "micro": "^9.0.0",
```
Comment (breath103) [inline package.json]: 얘는 cli에서만 사용되네요. devDependency로

Comment (mooyoul) [reply package.json]: 이거 devDependnecy로 빼고싶었는데 그러면 corgi를 의존성으로 가지는 패키지들에서 corgi 설치하면 evDependency들을 설치하지 않더라고요 ㅠㅠ

Comment (breath103) [reply package.json]: 음 그러면 차라리 peerDependency로 해도 되고요. 여튼 corgi는 결국 lambda에 올라가는 시스템인데 여기에 테스트 환경에서 사용할 webserver framework를 넣는건 안될 것 같아요
1) peer로 빼던지
2) 이 기능 자체를 vingle-corgi-local로 분리하던지
해서 정리하고 머지할게요

Comment (mooyoul) [reply package.json]: peer로 뽑으면 아마 프로젝트에서 npm install 할때 unmet peer dependency 에러가 올라올테니,
패키지를 따로 vingle-corgi-local 로 분리하고, 분리한 패키지쪽에서 corgi를 peer dependency로 가지게 할게요!

**Comment by mooyoul** [comment]
We've discussed about package dependency, since currently we have to install `micro` package even developers doesn't use this local server feature, we worried about that optional package dependency could make impacts to lambda deployment builds (e.g. increases deployment size, longer CI build time, etc.)  
So we decided to split this feature as individual package like `vingle-corgi-local`, which has `vingle-corgi` as peerDependency.

**Pull request closed without merging**